src/test/javascript/spec/component/overview/course-discussion/course-discussion.component.spec.ts

```diff
@@ -0,0 +1,373 @@
+import {
+    metisAnswerPostUser1,
+    metisCourse,
+    metisCoursePosts,
+    metisCoursePostsWithCourseWideContext,
+    metisExercise,
+    metisExercisePosts,
+    metisLecture,
+    metisLecturePosts,
+    metisPostExerciseUser1,
+    metisPostExerciseUser2,
+    metisPostLectureUser1,
+    metisPostLectureUser2,
+    metisUpVoteReactionUser1,
+} from '../../../helpers/sample/metis-sample-data';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('CourseDiscussionComponent', () => {
+    let component: CourseDiscussionComponent;
+    let fixture: ComponentFixture<CourseDiscussionComponent>;
+    let courseManagementService: CourseManagementService;
+    let metisService: MetisService;
+    let metisServiceSpy: SinonSpy;
+    let post1: Post;
+    let post2: Post;
+    let post3: Post;
+    let post4: Post;
+    let posts: Post[];
+
+    beforeEach(() => {
+        const id = metisCourse.id;
+        const parentRoute = {
+            params: of({ id }),
+        } as any as ActivatedRoute;
+        const route = { parent: parentRoute } as any as ActivatedRoute;
+
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, HttpClientTestingModule, MockModule(FormsModule), MockModule(ReactiveFormsModule)],
+            providers: [
+                FormBuilder,
+                { provide: ExerciseService, useClass: MockExerciseService },
+                { provide: AnswerPostService, useClass: MockAnswerPostService },
+                { provide: PostService, useClass: MockPostService },
+                { provide: AccountService, useClass: MockAccountService },
+                { provide: ActivatedRoute, useValue: route },
+            ],
+            declarations: [
+                CourseDiscussionComponent,
+                MockComponent(PostingsThreadComponent),
+                MockComponent(PostCreateEditModalComponent),
+                MockPipe(ArtemisTranslatePipe),
+                MockDirective(NgbTooltip),
+                MockComponent(ButtonComponent),
+            ],
+        })
+            .overrideComponent(PageDiscussionSectionComponent, {
+                set: {
+                    providers: [{ provide: MetisService, useClass: MetisService }],
+                },
+            })
+            .compileComponents()
+            .then(() => {
+                courseManagementService = TestBed.inject(CourseManagementService);
+                stub(courseManagementService, 'findOneForDashboard').returns(of({ body: metisCourse }) as Observable<HttpResponse<Course>>);
+                fixture = TestBed.createComponent(CourseDiscussionComponent);
+                component = fixture.componentInstance;
+                component = fixture.componentInstance;
+                metisService = fixture.debugElement.injector.get(MetisService);
+                metisServiceSpy = spy(metisService, 'getFilteredPosts');
+            });
+    });
+
+    afterEach(function () {
```

Comment:
```suggestion
    afterEach(() => {
```
to keep the style similar to the others

Reply:
:white_check_mark: